tools/flaky-test-reporter/slack_notification.go

```diff
@@ -107,25 +102,29 @@ func (c *SlackClient) writeSlackMessage(text, channel string) error {
 // createSlackMessageForRepo creates slack message layout from RepoData
 func createSlackMessageForRepo(rd *RepoData, flakyIssuesMap map[string][]*flakyIssue) string {
 	flakyTests := getFlakyTests(rd)
-	message := fmt.Sprintf("As of %s, there are %d flaky tests in '%s'",
-		time.Unix(*rd.LastBuildStartTime, 0).String(), len(flakyTests), rd.Config.Repo)
+	message := fmt.Sprintf("As of %s, there are %d flaky tests in '%s' from repo '%s'",
+		time.Unix(*rd.LastBuildStartTime, 0).String(), len(flakyTests), rd.Config.Name, rd.Config.Repo)
 	flakyRate := getFlakyRate(rd)
 	if flakyRate > threshold { // Don't list each test as this can be huge
 		message += fmt.Sprintf("\n>- skip displaying all tests as flaky rate above '%0.2f%%'", threshold)
-		if flakyIssues, ok := flakyIssuesMap[getBulkIssueIdentity(rd, flakyRate)]; ok {
-			// When flaky rate is above threthold, there is only one issue created,
+		if flakyIssues, ok := flakyIssuesMap[getBulkIssueIdentity(rd, flakyRate)]; ok && rd.Config.PostIssue {
+			// When flaky rate is above threshold, there is only one issue created,
 			// so there is only one element in flakyIssues
 			for _, fi := range flakyIssues {
 				message += fmt.Sprintf("\t%s", fi.issue.GetHTMLURL())
 			}
+		} else {
+			message += fmt.Sprintf("\t Job is marked to not create GitHub issues")
 		}
 	} else {
 		for _, testFullName := range flakyTests {
 			message += fmt.Sprintf("\n>- %s", testFullName)
-			if flakyIssues, ok := flakyIssuesMap[getIdentityForTest(testFullName, rd.Config.Repo)]; ok {
+			if flakyIssues, ok := flakyIssuesMap[getIdentityForTest(testFullName, rd.Config.Repo)]; ok && rd.Config.PostIssue {
 				for _, fi := range flakyIssues {
 					message += fmt.Sprintf("\t%s", fi.issue.GetHTMLURL())
 				}
+			} else {
+				message += fmt.Sprintf("\t Job is marked to not create GitHub issues")
```

Comment:
should be my last comment, I would expect this line show up only once instead of multiple times for each message. You can also combine this line with line 117 if only show this line once

Reply:
To make it clear, the Slack message for with or w/o PostIssue flag should be almost identical. The only difference here is:
- (PostIssue = true) There is a github issue link for each failed test
- (PostIssue = false) Additional message indicates "Job is marked to not create GitHub issues"

An example message for PostIssue=false in my mind looks like this:
```
As of YYYY-MM-DD, there are 2 flaky tests in 'ci-knative-serving-istio-1.0.7-mesh' in repo 'serving' (Job is marked to not create GitHub issues):
>- TestNameA
>- TestNameB
See Testgrid for up-to-date flaky tests information: [TESTGRID_URL]
```

You don't have to follow exactly the same format, the idea is we should deliver same amount of information no matter PostIssue is true or false

Reply:
I understand now, thank's for the clarification. Next commit will fix this.